### ISSUE TITLE
HTTPS OTA example Failed to verify peer certificate

### ISSUE THREAD

**Issue by gomez9656** [espressif/arduino-esp32#7626]
### Board

ESP32 Dev Module

### Device Description

ESP32 board powered by PC USB

### Hardware Configuration

Out of the box HPPTS OTA example for ESP32 in Arduino https://github.com/espressif/arduinoesp32/blob/master/libraries/Update/examples/HTTPS_OTA_Update/HTTPS_OTA_Update.ino  e

### Version

latest master (checkout manually)

### IDE Name

PlatformIO with Arduino framework

### Operating System

Windows 11

### Flash frequency

40MHz

### PSRAM enabled

yes

### Upload speed

115200

### Description

Hello there!

I'm working with the HTTPS_OTA_Update example in my ESP32 board but I get the following error:
![image](https://user-images.githubusercontent.com/59805687/209448709-e76193a6-081f-4f15-8c9b-e64ae7624649.png)

The HTTPS server is running in git bash with the OTA ESP-IDF example instructions: https://github.com/espressif/esp-idf/tree/master/examples/system/ota

I tested this example with HTTP + python server and it works.  THE AWS S3 example doesn't work because the code is for HTTP, and now the AWS S3 bucket is HTTPS.

I also tested the example using echo "" | openssl s_client -showcerts -connect raw.githubusercontent.com:443 | sed -n "1,/Root/d; /BEGIN/,/END/p" | openssl x509 -outform PEM >ca_cert.pem and storing the firmware in my github: https://github.com/gomez9656/aaaaaa/blob/master/firmware.bin

This option gives me the error: 
![image](https://user-images.githubusercontent.com/59805687/209448763-8c2c6ca1-6196-4a3a-af40-6593afc4ddd8.png)

Overall. when using a web server I get a certificate error, and when using the local HTTPS server I got the first error.

What should I be looking for here? 

I really appreciate all the help and thanks in advance :) + merry Xmas 24th December 👍 

### Sketch

```cpp
Same example as https://github.com/espressif/arduino-esp32/blob/master/libraries/Update/examples/HTTPS_OTA_Update/HTTPS_OTA_Update.ino

I have also used ESP-IDF example: https://github.com/espressif/esp-idf/tree/master/examples/system/ota/advanced_https_ota
```

### Debug Message

```plain
When using PlatformIO + HTTPS local server in git bash
I (852) wifi:mode : sta (c4:de:e2:12:0a:64)
I (852) wifi:enable tsf
␛[0;32mI (862) example_connect: Waiting for IP(s)␛[0m
I (3262) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:1
I (4792) wifi:state: init -> auth (b0)
I (4792) wifi:state: auth -> assoc (0)
I (4792) wifi:state: assoc -> run (10)
I (4902) wifi:connected with TIGO-8E48, aid = 2, channel 1, BW20, bssid = 6c:63:9c:3d:37:28
I (4902) wifi:security: WPA2-PSK, phy: bgn, rssi: -51
I (4902) wifi:pm start, type: 1

I (4952) wifi:AP's beacon interval = 102400 us, DTIM period = 1
␛[0;32mI (6142) esp_netif_handlers: example_connect: sta ip: 192.168.1.18, mask: 255.255.255.0, gw: 192.168.1.254␛[0m
␛[0;32mI (6142) example_connect: Got IPv4 event: Interface "example_connect: sta" address: 192.168.1.18␛[0m
␛[0;32mI (6642) example_connect: Got IPv6 event: Interface "example_connect: sta" address: fe80:0000:0000:0000:c6de:e2ff:fe12:0a64, type: ESP_IP6_ADDR_IS_LINK_LOCAL␛[0m
␛[0;32mI (6642) example_connect: Connected to example_connect: sta␛[0m
␛[0;32mI (6652) example_connect: - IPv4 address: 192.168.1.18␛[0m
␛[0;32mI (6652) example_connect: - IPv6 address: fe80:0000:0000:0000:c6de:e2ff:fe12:0a64, type: ESP_IP6_ADDR_IS_LINK_LOCAL␛[0m
I (6672) wifi:Set ps type: 0
␛[0;32mI (6672) advanced_https_ota_example: Starting Advanced OTA example␛[0m
W (6722) wifi:<ba-add>idx:0 (ifx:0, 6c:63:9c:3d:37:28), tid:0, ssn:54, winSize:64
␛[0;32mI (6732) mbedtls: ssl_tls.c:5904 => handshake
␛[0m
␛[0;32mI (6732) mbedtls: ssl_cli.c:4483 client state: 0
␛[0m
␛[0;32mI (6732) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (6742) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (6742) mbedtls: ssl_cli.c:4483 client state: 1
␛[0m
␛[0;32mI (6752) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (6762) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (6762) mbedtls: ssl_cli.c:999 => write client hello
␛[0m
␛[0;32mI (6772) mbedtls: ssl_msg.c:2542 => write handshake message
␛[0m
␛[0;32mI (6772) mbedtls: ssl_msg.c:2701 => write record
␛[0m
␛[0;32mI (6782) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (6782) mbedtls: ssl_msg.c:2122 message length: 222, out_left: 222
␛[0m
␛[0;32mI (6792) mbedtls: ssl_msg.c:2127 ssl->f_send() returned 222 (-0xffffff22)
␛[0m
␛[0;32mI (6802) mbedtls: ssl_msg.c:2155 <= flush output
␛[0m
␛[0;32mI (6802) mbedtls: ssl_msg.c:2870 <= write record
␛[0m
␛[0;32mI (6812) mbedtls: ssl_msg.c:2678 <= write handshake message
␛[0m
␛[0;32mI (6822) mbedtls: ssl_cli.c:1467 <= write client hello
␛[0m
␛[0;32mI (6822) mbedtls: ssl_cli.c:4483 client state: 2
␛[0m
␛[0;32mI (6832) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (6832) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (6842) mbedtls: ssl_cli.c:2082 => parse server hello
␛[0m
␛[0;32mI (6842) mbedtls: ssl_msg.c:3941 => read record
␛[0m
␛[0;32mI (6852) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (6862) mbedtls: ssl_msg.c:2043 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (6862) mbedtls: ssl_msg.c:2068 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (6872) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
␛[0m
␛[0;32mI (6882) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (6882) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (6892) mbedtls: ssl_msg.c:2043 in_left: 5, nb_want: 70
␛[0m
␛[0;32mI (6892) mbedtls: ssl_msg.c:2068 in_left: 5, nb_want: 70
␛[0m
␛[0;32mI (6902) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 65 (-0xffffffbf)
␛[0m
␛[0;32mI (6912) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (6912) mbedtls: ssl_msg.c:4015 <= read record
␛[0m
␛[0;32mI (6922) mbedtls: ssl_cli.c:2385 server hello, total extension length: 21
␛[0m
␛[0;32mI (6932) mbedtls: ssl_cli.c:2627 <= parse server hello
␛[0m
␛[0;32mI (6932) mbedtls: ssl_cli.c:4483 client state: 3
␛[0m
␛[0;32mI (6942) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (6942) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (6952) mbedtls: ssl_tls.c:2878 => parse certificate
␛[0m
␛[0;32mI (6952) mbedtls: ssl_msg.c:3941 => read record
␛[0m
␛[0;32mI (6962) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (6962) mbedtls: ssl_msg.c:2043 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (6972) mbedtls: ssl_msg.c:2068 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (6982) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
␛[0m
␛[0;32mI (6982) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (6992) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (7002) mbedtls: ssl_msg.c:2043 in_left: 5, nb_want: 992
␛[0m
␛[0;32mI (7002) mbedtls: ssl_msg.c:2068 in_left: 5, nb_want: 992
␛[0m
␛[0;32mI (7012) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 987 (-0xfffffc25)
␛[0m
␛[0;32mI (7022) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (7032) mbedtls: ssl_msg.c:4015 <= read record
␛[0m
␛[0;32mI (7042) mbedtls: ssl_tls.c:2989 <= parse certificate
␛[0m
␛[0;32mI (7042) mbedtls: ssl_cli.c:4483 client state: 4
␛[0m
␛[0;32mI (7042) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (7042) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (7052) mbedtls: ssl_cli.c:3094 => parse server key exchange
␛[0m
␛[0;32mI (7062) mbedtls: ssl_msg.c:3941 => read record
␛[0m
␛[0;32mI (7062) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (7072) mbedtls: ssl_msg.c:2043 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (7072) mbedtls: ssl_msg.c:2068 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (7082) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
␛[0m
␛[0;32mI (7092) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (7092) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (7102) mbedtls: ssl_msg.c:2043 in_left: 5, nb_want: 406
␛[0m
␛[0;32mI (7112) mbedtls: ssl_msg.c:2068 in_left: 5, nb_want: 406
␛[0m
␛[0;32mI (7112) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 401 (-0xfffffe6f)
␛[0m
␛[0;32mI (7122) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (7132) mbedtls: ssl_msg.c:4015 <= read record
␛[0m
␛[0;32mI (7132) mbedtls: ssl_cli.c:2699 ECDH curve: secp521r1
␛[0m
␛[0;32mI (7142) mbedtls: ssl_cli.c:3018 Server used SignatureAlgorithm 1
␛[0m
␛[0;32mI (7142) mbedtls: ssl_cli.c:3020 Server used HashAlgorithm 6
␛[0m
␛[0;32mI (7182) mbedtls: ssl_cli.c:3487 <= parse server key exchange
␛[0m
␛[0;32mI (7182) mbedtls: ssl_cli.c:4483 client state: 5
␛[0m
␛[0;32mI (7182) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (7192) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (7192) mbedtls: ssl_cli.c:3522 => parse certificate request
␛[0m
␛[0;32mI (7202) mbedtls: ssl_msg.c:3941 => read record
␛[0m
␛[0;32mI (7202) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (7212) mbedtls: ssl_msg.c:2043 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (7222) mbedtls: ssl_msg.c:2068 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (7222) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
␛[0m
␛[0;32mI (7232) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (7232) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (7242) mbedtls: ssl_msg.c:2043 in_left: 5, nb_want: 9
␛[0m
␛[0;32mI (7252) mbedtls: ssl_msg.c:2068 in_left: 5, nb_want: 9
␛[0m
␛[0;32mI (7252) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 4 (-0xfffffffc)
␛[0m
␛[0;32mI (7262) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (7272) mbedtls: ssl_msg.c:4015 <= read record
␛[0m
␛[0;32mI (7272) mbedtls: ssl_cli.c:3678 <= parse certificate request
␛[0m
␛[0;32mI (7282) mbedtls: ssl_cli.c:4483 client state: 6
␛[0m
␛[0;32mI (7282) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (7292) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (7292) mbedtls: ssl_cli.c:3689 => parse server hello done
␛[0m
␛[0;32mI (7302) mbedtls: ssl_msg.c:3941 => read record
␛[0m
␛[0;32mI (7312) mbedtls: ssl_msg.c:4011 reuse previously read message
␛[0m
␛[0;32mI (7312) mbedtls: ssl_msg.c:4015 <= read record
␛[0m
␛[0;32mI (7322) mbedtls: ssl_cli.c:3719 <= parse server hello done
␛[0m
␛[0;32mI (7322) mbedtls: ssl_cli.c:4483 client state: 7
␛[0m
␛[0;32mI (7332) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (7342) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (7342) mbedtls: ssl_tls.c:2229 => write certificate
␛[0m
␛[0;32mI (7352) mbedtls: ssl_tls.c:2243 <= skip write certificate
␛[0m
␛[0;32mI (7352) mbedtls: ssl_cli.c:4483 client state: 8
␛[0m
␛[0;32mI (7362) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (7362) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (7372) mbedtls: ssl_cli.c:3734 => write client key exchange
␛[0m
␛[0;32mI (8812) mbedtls: ssl_msg.c:2542 => write handshake message
␛[0m
␛[0;32mI (8812) mbedtls: ssl_msg.c:2701 => write record
␛[0m
␛[0;32mI (8812) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (8812) mbedtls: ssl_msg.c:2122 message length: 143, out_left: 143
␛[0m
␛[0;32mI (8822) mbedtls: ssl_msg.c:2127 ssl->f_send() returned 143 (-0xffffff71)
␛[0m
␛[0;32mI (8832) mbedtls: ssl_msg.c:2155 <= flush output
␛[0m
␛[0;32mI (8832) mbedtls: ssl_msg.c:2870 <= write record
␛[0m
␛[0;32mI (8842) mbedtls: ssl_msg.c:2678 <= write handshake message
␛[0m
␛[0;32mI (8852) mbedtls: ssl_cli.c:4151 <= write client key exchange
␛[0m
␛[0;32mI (8852) mbedtls: ssl_cli.c:4483 client state: 9
␛[0m
␛[0;32mI (8862) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (8862) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (8872) mbedtls: ssl_cli.c:4196 => write certificate verify
␛[0m
␛[0;32mI (8872) mbedtls: ssl_tls.c:1728 => derive keys
␛[0m
␛[0;32mI (8882) mbedtls: ssl_tls.c:1979 => calc verify sha384
␛[0m
␛[0;32mI (8892) mbedtls: ssl_tls.c:1987 <= calc verify
␛[0m
␛[0;32mI (8912) mbedtls: ssl_tls.c:1807 <= derive keys
␛[0m
␛[0;32mI (8912) mbedtls: ssl_cli.c:4221 <= skip write certificate verify
␛[0m
␛[0;32mI (8912) mbedtls: ssl_cli.c:4483 client state: 10
␛[0m
␛[0;32mI (8922) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (8922) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (8932) mbedtls: ssl_msg.c:5012 => write change cipher spec
␛[0m
␛[0;32mI (8932) mbedtls: ssl_msg.c:2542 => write handshake message
␛[0m
␛[0;32mI (8942) mbedtls: ssl_msg.c:2701 => write record
␛[0m
␛[0;32mI (8942) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (8952) mbedtls: ssl_msg.c:2122 message length: 6, out_left: 6
␛[0m
␛[0;32mI (8962) mbedtls: ssl_msg.c:2127 ssl->f_send() returned 6 (-0xfffffffa)
␛[0m
␛[0;32mI (8962) mbedtls: ssl_msg.c:2155 <= flush output
␛[0m
␛[0;32mI (8972) mbedtls: ssl_msg.c:2870 <= write record
␛[0m
␛[0;32mI (8982) mbedtls: ssl_msg.c:2678 <= write handshake message
␛[0m
␛[0;32mI (8982) mbedtls: ssl_msg.c:5026 <= write change cipher spec
␛[0m
␛[0;32mI (8992) mbedtls: ssl_cli.c:4483 client state: 11
␛[0m
␛[0;32mI (8992) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (9002) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (9002) mbedtls: ssl_tls.c:3533 => write finished
␛[0m
␛[0;32mI (9012) mbedtls: ssl_tls.c:3399 => calc  finished tls sha384
␛[0m
␛[0;32mI (9022) mbedtls: ssl_tls.c:3437 <= calc  finished
␛[0m
␛[0;32mI (9022) mbedtls: ssl_msg.c:2542 => write handshake message
␛[0m
␛[0;32mI (9032) mbedtls: ssl_msg.c:2701 => write record
␛[0m
␛[0;32mI (9032) mbedtls: ssl_msg.c:632 => encrypt buf
␛[0m
␛[0;32mI (9042) mbedtls: ssl_msg.c:1120 <= encrypt buf
␛[0m
␛[0;32mI (9052) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (9052) mbedtls: ssl_msg.c:2122 message length: 45, out_left: 45
␛[0m
␛[0;32mI (9062) mbedtls: ssl_msg.c:2127 ssl->f_send() returned 45 (-0xffffffd3)
␛[0m
␛[0;32mI (9062) mbedtls: ssl_msg.c:2155 <= flush output
␛[0m
␛[0;32mI (9072) mbedtls: ssl_msg.c:2870 <= write record
␛[0m
␛[0;32mI (9082) mbedtls: ssl_msg.c:2678 <= write handshake message
␛[0m
␛[0;32mI (9082) mbedtls: ssl_tls.c:3642 <= write finished
␛[0m
␛[0;32mI (9092) mbedtls: ssl_cli.c:4483 client state: 12
␛[0m
␛[0;32mI (9092) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (9102) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (9102) mbedtls: ssl_cli.c:4365 => parse new session ticket
␛[0m
␛[0;32mI (9112) mbedtls: ssl_msg.c:3941 => read record
␛[0m
␛[0;32mI (9122) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (9122) mbedtls: ssl_msg.c:2043 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (9132) mbedtls: ssl_msg.c:2068 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (9132) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
␛[0m
␛[0;32mI (9142) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (9152) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (9152) mbedtls: ssl_msg.c:2043 in_left: 5, nb_want: 175
␛[0m
␛[0;32mI (9162) mbedtls: ssl_msg.c:2068 in_left: 5, nb_want: 175
␛[0m
␛[0;32mI (9172) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 170 (-0xffffff56)
␛[0m
␛[0;32mI (9172) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (9182) mbedtls: ssl_msg.c:4015 <= read record
␛[0m
␛[0;32mI (9182) mbedtls: ssl_cli.c:4467 <= parse new session ticket
␛[0m
␛[0;32mI (9192) mbedtls: ssl_cli.c:4483 client state: 12
␛[0m
␛[0;32mI (9202) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (9202) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (9212) mbedtls: ssl_msg.c:5035 => parse change cipher spec
␛[0m
␛[0;32mI (9212) mbedtls: ssl_msg.c:3941 => read record
␛[0m
␛[0;32mI (9222) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (9232) mbedtls: ssl_msg.c:2043 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (9232) mbedtls: ssl_msg.c:2068 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (9242) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
␛[0m
␛[0;32mI (9252) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (9252) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (9262) mbedtls: ssl_msg.c:2043 in_left: 5, nb_want: 6
␛[0m
␛[0;32mI (9262) mbedtls: ssl_msg.c:2068 in_left: 5, nb_want: 6
␛[0m
␛[0;32mI (9272) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 1 (-0xffffffff)
␛[0m
␛[0;32mI (9282) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (9282) mbedtls: ssl_msg.c:4015 <= read record
␛[0m
␛[0;32mI (9292) mbedtls: ssl_msg.c:5099 <= parse change cipher spec
␛[0m
␛[0;32mI (9292) mbedtls: ssl_cli.c:4483 client state: 13
␛[0m
␛[0;32mI (9302) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (9312) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (9312) mbedtls: ssl_tls.c:3659 => parse finished
␛[0m
␛[0;32mI (9322) mbedtls: ssl_tls.c:3399 => calc  finished tls sha384
␛[0m
␛[0;32mI (9332) mbedtls: ssl_tls.c:3437 <= calc  finished
␛[0m
␛[0;32mI (9332) mbedtls: ssl_msg.c:3941 => read record
␛[0m
␛[0;32mI (9332) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (9342) mbedtls: ssl_msg.c:2043 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (9352) mbedtls: ssl_msg.c:2068 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (9352) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
␛[0m
␛[0;32mI (9362) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (9372) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (9372) mbedtls: ssl_msg.c:2043 in_left: 5, nb_want: 45
␛[0m
␛[0;32mI (9382) mbedtls: ssl_msg.c:2068 in_left: 5, nb_want: 45
␛[0m
␛[0;32mI (9382) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 40 (-0xffffffd8)
␛[0m
␛[0;32mI (9392) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (9402) mbedtls: ssl_msg.c:1144 => decrypt buf
␛[0m
␛[0;32mI (9402) mbedtls: ssl_msg.c:1741 <= decrypt buf
␛[0m
␛[0;32mI (9412) mbedtls: ssl_msg.c:4015 <= read record
␛[0m
␛[0;32mI (9412) mbedtls: ssl_tls.c:3730 <= parse finished
␛[0m
␛[0;32mI (9422) mbedtls: ssl_cli.c:4483 client state: 14
␛[0m
␛[0;32mI (9422) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (9432) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (9432) mbedtls: ssl_cli.c:4594 handshake: done
␛[0m
␛[0;32mI (9442) mbedtls: ssl_cli.c:4483 client state: 15
␛[0m
␛[0;32mI (9452) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (9452) mbedtls: ssl_msg.c:2114 <= flush output
␛[0m
␛[0;32mI (9462) mbedtls: ssl_tls.c:5915 <= handshake
␛[0m
␛[0;32mI (9462) mbedtls: ssl_msg.c:5820 => write
␛[0m
␛[0;32mI (9472) mbedtls: ssl_msg.c:2701 => write record
␛[0m
␛[0;32mI (9472) mbedtls: ssl_msg.c:632 => encrypt buf
␛[0m
␛[0;32mI (9482) mbedtls: ssl_msg.c:1120 <= encrypt buf
␛[0m
␛[0;32mI (9482) mbedtls: ssl_msg.c:2102 => flush output
␛[0m
␛[0;32mI (9492) mbedtls: ssl_msg.c:2122 message length: 137, out_left: 137
␛[0m
␛[0;32mI (9502) mbedtls: ssl_msg.c:2127 ssl->f_send() returned 137 (-0xffffff77)
␛[0m
␛[0;32mI (9502) mbedtls: ssl_msg.c:2155 <= flush output
␛[0m
␛[0;32mI (9512) mbedtls: ssl_msg.c:2870 <= write record
␛[0m
␛[0;32mI (9512) mbedtls: ssl_msg.c:5848 <= write
␛[0m
␛[0;32mI (9522) mbedtls: ssl_msg.c:5399 => read
␛[0m
␛[0;32mI (9522) mbedtls: ssl_msg.c:3941 => read record
␛[0m
␛[0;32mI (9532) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (9532) mbedtls: ssl_msg.c:2043 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (9542) mbedtls: ssl_msg.c:2068 in_left: 0, nb_want: 5
␛[0m
␛[0;32mI (9552) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
␛[0m
␛[0;32mI (9562) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (9562) mbedtls: ssl_msg.c:1886 => fetch input
␛[0m
␛[0;32mI (9572) mbedtls: ssl_msg.c:2043 in_left: 5, nb_want: 16413
␛[0m
␛[0;32mI (9572) mbedtls: ssl_msg.c:2068 in_left: 5, nb_want: 16413
␛[0m
␛[0;32mI (9582) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 4315 (-0xffffef25)
␛[0m
␛[0;32mI (9592) mbedtls: ssl_msg.c:2068 in_left: 4320, nb_want: 16413
␛[0m
␛[0;32mI (9592) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 4320 (-0xffffef20)
␛[0m
␛[0;32mI (9602) mbedtls: ssl_msg.c:2068 in_left: 8640, nb_want: 16413
␛[0m
␛[0;32mI (9612) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 4320 (-0xffffef20)
␛[0m
␛[0;32mI (9622) mbedtls: ssl_msg.c:2068 in_left: 12960, nb_want: 16413
␛[0m
␛[0;32mI (9622) mbedtls: ssl_msg.c:2069 ssl->f_recv(_timeout)() returned 3453 (-0xfffff283)
␛[0m
␛[0;32mI (9632) mbedtls: ssl_msg.c:2089 <= fetch input
␛[0m
␛[0;32mI (9672) mbedtls: ssl_msg.c:1144 => decrypt buf
␛[0m
␛[0;32mI (9712) mbedtls: ssl_msg.c:1741 <= decrypt buf
␛[0m
␛[0;32mI (9742) mbedtls: ssl_msg.c:4015 <= read record
␛[0m
␛[0;32mI (9742) mbedtls: ssl_msg.c:5693 <= read
␛[0m
␛[0;32mI (9742) HTTP_CLIENT: Body received in fetch header state, 0x3ffb9649, 467␛[0m
␛[0;32mI (9752) esp_https_ota: Starting OTA...␛[0m
␛[0;32mI (9752) esp_https_ota: Writing to partition subtype 16 at offset 0x110000␛[0m
␛[0;32mI (9762) advanced_https_ota_example: Running firmware version: 1␛[0m
␛[0;31mE (9772) esp_https_ota: Mismatch chip id, expected 0, found 5␛[0m
␛[0;31mE (9772) advanced_https_ota_example: Complete data was not received.␛[0m
␛[0;31mE (9792) advanced_https_ota_example: ESP_HTTPS_OTA upgrade failed␛[0m
```

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by gomez9656** [comment]
Hello again!

After long hours of debugging sessions, I finally solved the issue.

I had to set *config.skip_cert_common_name_check* variable to false using *HttpsOTA.begin(url, server_certificate, false)*

This variable was initialized to true when the class was initiated 
![image](https://user-images.githubusercontent.com/59805687/209717520-014ba1c9-c0e2-46ab-8453-e81e0472fb8d.png)

I double-checked with the ESP-IDF advanced_https_ota example and the behavior is the same. If I enable *example configuration -> skip server certificate CN fielcheck* then OTA doesn't work.
If I leave it unchecked, then it works. 

Altough the code is working for me, my final question is:

Why is *config.skip_cert_common_name_check*  initialized to true? shouldn't be false by default?

Thanks a lot :)
